### ISSUE TITLE
Add Grok Support

### ISSUE THREAD

**Issue by Luca-Blight** [pydantic/pydantic-ai#563]
It is very unique model in the sense that it has real-time access to content from X. 
I currently need it for a "social media expert" agent. I speculate that it will also be one of the top 3 closed source models within the next year.

**Comment by Luca-Blight** [comment]
Actually I can work on this if that's cool? May need to discuss some of the process with someone on your team. On that note I think it would be useful to add a "discussions" tab to the github repo.

**Comment by sydney-runkle** [comment]
> add a "discussions" tab to the github repo

done :)

**Comment by samuelcolvin** [comment]
Please let's not enable discussions, they just cause confusion. We already have slack and GitHub issue.

Happy to accept a PR to add grok, what questions do you have.

**Comment by Luca-Blight** [comment]
Going to explore this a bit and will circle back with any questions. 
You should expect a PR to appear within the next few days.

**Comment by samuelcolvin** [comment]
See also #261.

I think we can just use the OpenAI model and document it.

**Comment by Luca-Blight** [comment]
Agreed, I saw the compatibility on their website as well. Where is best to document it?
